docs/test-plan/stream-subscriber/stream-subscriber-e2e-test-plan.md

```diff
@@ -0,0 +1,912 @@
+# `Stream Subscriber Plugin` Test Plan
+
+## Overview
+
+The `Stream Subscriber Plugin` allows clients to subscribe (issue a request for
+gRPC server streaming) to a Block Node and receive a stream of Blocks based on
+the specified request criteria.
+
+Blocks could be streamed from the Block Node's history and/or from the live
+Block Stream as new Blocks are being published to the Block Node.
+
+## Key Considerations
+
+- **SubscriberServicePlugin**: The plugin accepts gRPC requests from clients
```

Comment:
Good call out.
Would be good to link to the API so it's clear what users are hitting